promql/engine.go

```diff
@@ -700,16 +726,21 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 	enh := &EvalNodeHelper{out: make(Vector, 0, biggestLen)}
 	seriess := make(map[uint64]Series, biggestLen) // Output series by series hash.
 	for ts := ev.startTimestamp; ts <= ev.endTimestamp; ts += ev.interval {
+		// Reset num. of samples in memory after each timestamp.
+		tempNumSamples = ev.currentSamples
```

Comment:
You're losing the original value of tempNumSamples here

Reply:
You're still losing this here, you need a different variable to store this in.

Reply:
I see what you're saying. The number of samples we have going into the function is the value we want to ues (plus the amount of new samples) when returning, but we want to reset to the previous value of `ev.currentSamples` after each timestamp iteration. Latest commit should fix this.